variables.tf

```diff
@@ -1,4 +1,183 @@
-variable "example" {
-  description = "Example variable"
-  default     = "hello world"
+variable "allowed_uses" {
+  description = <<-EOT
+  List of keywords each describing a use that is permitted for the issued certificate.
+  Must be one of of the values outlined in [self_signed_cert.allowed_uses](https://registry.terraform.io/providers/hashicorp/tls/latest/docs/resources/self_signed_cert#allowed_uses).
+  EOT
+  type        = list(string)
+}
+
+variable "basic_constraints" {
```

Comment:
@korenyoni if in the future there going to be more attributes in `basic_constraints`, then maybe we can add the same var to all the tests instead of using `is_ca` var?

Reply:
in general, we just copy (almost) all vars from the module to the tests, and customize them in TF defaults